### PR TITLE
add table view for the enterprise contract results

### PR DESCRIPTION
**Pull request by sahil143**
## Fixes 

https://issues.redhat.com/browse/HAC-3424

## Description

add Enterprise contract result under security tab in pipeline runs

[TODO]:

- []  ~Update the label to identify the Enterprise contract pipeline run (https://github.com/openshift/hac-dev/pull/530/files#diff-9eaec787536555e3f629da6787ae72f774c60541223c569ec9a582d6067e0222R71)~
Created issue https://issues.redhat.com/browse/HAC-3644 to update the label once https://github.com/redhat-appstudio/build-definitions/pull/369 gets merged
- [x] Add Security Tab on task run details page
- [x] Add unit tests

## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

image

image
image

Task Run Security Tab
image

## Browser conformance: 

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by christianvogt** [comment]
Will we be adding this tab to the EC task details page as well?
cc @MariaLeonova

**Comment by MariaLeonova** [comment]
@sahil143 this is great! Can you please also add a screenshot of an expanded section? 
The Date found filed is not populated - what are the reasons there? Will it never be populated or could it be fixed?

**Comment by sahil143** [comment]
@MariaLeonova added another screenshot in the description. we get a date in case of failure, not in case of success.

**Comment by sahil143** [comment]
> Thank you very much! It looks great! Regarding the date: can I ask, out of curiosity, where it comes from? Why is Jan 1, 2022 in this case?

It comes from Enterprise contract logs only.

**Comment by sahil143** [comment]
Security tab on Taskrun details page
image

**Comment by mattreid** [comment]
@MariaLeonova @sahil143 You should consider renaming the date column. In the log, that's effective_on, which is supposed to tell the user the date when that rule started being enforced. Date found, as we call it instead in the UI, implies to me that's when the failure was discovered, and would be more or less the same for every warning and failure coming back within this pipeline run's contract evaluation.

**Comment by MariaLeonova** [comment]
@mattreid you're right - that's what I supposed it would be (found date - when the pipeline run happened). And if it isn't, we need to call that out in the UI. Or, possibly, is this is the case, we don't need this column at all? What value does it bring?

**Comment by zregvart** [comment]
> Or, possibly, is this is the case, we don't need this column at all? What value does it bring?

When new policy rules are introduced an enforcement date (`effective_on`) can be set in the future, say a month from when the rule is defined. So within that month a warning will be reported but after month a has passed this rule will result in failures. I think it is useful for folk to know when such rules will be in effect so they can adapt and implement any changes needed.

**Comment by zregvart** [comment]
> @zregvart thank you for explaining that! What would the status of such a rule be then? Currently we have 'failed', 'warning' and 'passed'

I'd still call it a warning, I would not try to do to much interpretation of the data in the UI, if we do then we tie the behavior of the UI closely to the behavior of EC. For example say that the requirement changes that a week before the `effective_on` date instead of reporting a warning EC needs to report failure instead.

> I would suggest we label the column 'Effective from' and call out the future dates specifically, maybe mark by color?

:+1: I think that would be a good approach.

**Comment by sahil143** [comment]
Updated column name 
image

**Comment by christianvogt** [comment]
Expanded rows should have the blue treatment on both the original row and the expansion row:
image
https://patternfly-react.surge.sh/components/table#composable-expandable

**Comment by MariaLeonova** [comment]
@zregvart thank you for reviewing! I see it as an opportunity to do more work on this column in the future, as well as possibly incorporate notifications for users.

**Review by karthikjeeyar — COMMENTED** [review]
@MariaLeonova I think we should be showing `view logs` even when you expand the graph, wdyt?

Screenshot 2023-04-05 at 3 42 13 PM

File: src/components/ApplicationDetails/tabs/overview/visualization/utils/node-utils.ts [hunk 1]
```diff
@@ -203,6 +205,7 @@ export const resourceToPipelineNode = (
     workflowType,
     status,
     resources: [resource],
+    pipelineRun,
```
[the remark this answers is not in the document]
Comment (sahil143) [reply src/components/ApplicationDetails/tabs/overview/visualization/utils/node-utils.ts:208]: This is being used in Node tooltip

File: src/components/ApplicationDetails/tabs/overview/visualization/hooks/useAppApplicationTestNodes.ts [hunk 2]
```diff
@@ -60,6 +60,8 @@ export const useAppApplicationTestNodes = (
             WorkflowNodeType.APPLICATION_TEST,
             previousTasks,
             testPipeline ? pipelineRunStatus(testPipeline) : runStatus.Pending,
+            undefined,
+            testPipeline,
```
[the remark this answers is not in the document]
Comment (sahil143) [reply src/components/ApplicationDetails/tabs/overview/visualization/hooks/useAppApplicationTestNodes.ts:64]: same as above

**Comment by MariaLeonova** [comment]
> @MariaLeonova I think we should be showing `view logs` even when you expand the graph, wdyt?
> 
> Screenshot 2023-04-05 at 3 42 13 PM

agreed

**Pull request merged**